### ISSUE TITLE
[P1] Type mismatches in struct/union initializers not diagnosed

### ISSUE THREAD

**Issue by ChaseWNorton** [anthropics/claudes-c-compiler#86]
## Summary

The compiler silently accepts struct initializers where element types don't match the field types. For example, a string literal initializing an `int` field produces no warning.

## Reproduction

```c
struct { int x, y, z; } s = { "oops", 0, 0 };
```

```bash
ccc -c test.c -o /dev/null
# No warning or error — should diagnose type mismatch
```

GCC output:
```
warning: initialization of 'int' from 'char *' makes integer from pointer without a cast
```

Also applies to excess elements in struct initializers:
```c
struct { int x, y, z; } s = { "oops", 0, &s, 42 };
// "oops" for int, &s for int, and 42 is a 4th element for a 3-field struct
```

## Root cause

`src/frontend/sema/analysis.rs` — the initializer analysis (~line 656-675) only type-checks scalar `Initializer::Expr` cases. It doesn't validate type compatibility between initializer elements and their corresponding struct/union fields within `Initializer::List`.

## Expected behavior

Each element in a struct/union initializer should be type-checked against the corresponding field type. Incompatible types (e.g., `char *` → `int`) should produce a warning matching GCC/Clang behavior.

## Acceptance criteria

- [ ] `struct { int x; } s = { "hello" }` produces a type mismatch warning
- [ ] `struct { int *p; } s = { 42 }` produces an integer-to-pointer warning
- [ ] Valid initializers still compile cleanly
- [ ] Excess elements in struct initializer are diagnosed
- [ ] Test using `sema_warning_count()` / `sema_error_count()`

## Metadata

- **Reporter**: @lephe in #57
- **Phase**: Sema
- **Decomposed from**: #57

**Comment by ChaseWNorton** [comment]
**Triaged** — validated during decomposition of parent issue. Root cause identified, reproduction case provided, acceptance criteria defined. Ready for pickup.

**Comment by ChaseWNorton** [comment]
Part of [MILESTONE] M7: Sema Hardening — Type Safety at Scale (#104)
